g3doc/obtaining-training-data.md

```diff
@@ -0,0 +1,50 @@
+# Obtaining Training Data
+
+This document gives an overview of how to use the Gematria tooling to
+parse/process training data from multiple sources.
+
+## Parsing training data from the BHive dataset
+
+The publically available [BHive](https://github.com/ithemal/bhive) dataset
+has a decent amount of training data available in CSV format. Gematria
+provides native tooling to process these CSV files into `.tfrecord` files
+that can be used for further processing.
+
+For example, to process BHive throughput data for the Intel Skylake
+microarchitecture, we can begin by downloading one of the BHive CSVs:
+
+```bash
+curl -L https://raw.githubusercontent.com/ithemal/bhive/5f1d50077ac0779fd227b261dcf517862c7104bd/benchmark/throughput/skl.csv > skl.csv
+```
+
+If you look at the file, you'll notice that every line is formatted as
+`<machine code of BB in hex>,<throughput in cycles>>`. This is referred to as
+the BHive format. There are other projects composing similar datasets (such
+as Adreas Abel's [uica-eval](https://github.com/andreas-abel/uiCA-eval) which
+contains several improvements over the original BHive dataset) that use similar
+but not necessarily the same CSV format. Note that Gematria only supports the
+BHive format outlined above, so some preprocessing might be required depending
+upon the input data format.
```

Comment:
I'd put this to a subsection at the end of the BHive section. Something like

```
## Other CSV-based formats

Other projects, such as Andreas Abel's [uica-eval](...) use a similar format. However, there are subtle differences between the formats, and additional pre-processing may be needed.
```

I didn't look at the data format in `uica-eval`, but if it is sufficiently close, adding an import tool might be less work than explaining why we don't support it directly 😄  I can check it out when I'm back from vacations.

Reply:
It's pretty close. The CSV format in `uica-eval` just has a couple extra columns. The main thing that I think would be needed would be flags to specify the column indices for the BB hex and throughput value. I should be able to get a patch up soon for that.